package/MDAnalysis/coordinates/GRO.py

```diff
@@ -352,21 +378,26 @@ def write(self, selection, frame=None):
                         resname=resname,
                         index=truncated_atom_index,
                         name=name,
-                        pos=coordinates[atom_index]
+                        pos=positions[atom_index]
                     ))
 
+            try:  # for AtomGroup/Universe
```

Comment:
You can move this try/except in to the `else` block below, so it's only created when needed